sympy/ntheory/residue_ntheory.py

```diff
@@ -742,6 +743,48 @@ def _nthroot_mod1(s, q, p, all_roots):
         return res
     return min(res)
 
+def _nthroot_mod_composite(a, n, m):
+    """
+    Find the solutions to ``x**n = a mod m`` when m is not prime.
+    """
+    from sympy.ntheory.modular import crt
+    f = factorint(m)
+    dd = {}
+    for p, e in f.items():
+        tot_roots = set()
+        if e == 1:
+            tot_roots.update(nthroot_mod(a, n, p, True) or [])
+        else:
+            for root in nthroot_mod(a, n, p, True) or []:
+                rootn = pow(root, n)
+                diff = (rootn // (root or 1) * n) % p
+                if diff != 0:
+                    ppow = p
+                    for j in range(1, e):
+                        ppow *= p
+                        root = (root - (rootn - a) * mod_inverse(diff, p)) % ppow
```

Comment:
This line should probably be modified.

Reply:
yes you are right, the rootn will change with change in root